common-react/@dbeaver/ui-kit/src/Combobox/Combobox.tsx

```diff
@@ -0,0 +1,191 @@
+  defaultValue?: string;
+  getInputFromItem?: (value: T) => string;
+}
+
+/**
+ * ComboboxInput - Basic input component that handles search value and blur logic
+ * Restores the selected value if the input value is not in the items list
+ * If only one item is available, it selects that item automatically
+ */
+export function ComboboxInput<T = unknown>({
+  onBlur,
+  onKeyDown,
+  defaultValue,
+  ref,
+  getInputFromItem,
+  ...props
+}: ComboboxInputProps<T> & { ref?: React.Ref<HTMLInputElement> }) {
+  const store = useComboboxContext();
+
+  function restoreInputValue() {
+    const { selectedValue, items, value } = store!.getState();
+
+    if (typeof selectedValue !== 'string' || (value === selectedValue && !getInputFromItem)) {
+      return;
+    }
+
+    const nextValue = (selectedValue || defaultValue) as string;
+
+    if (!nextValue) {
+      return;
+    }
+
+    if (getInputFromItem) {
+      const nextItem = (items?.find(item => item.value === nextValue) as ComboboxStoreState['items'][number] & { item: T })?.item;
+      if (nextItem) {
+        store!.setValue(getInputFromItem(nextItem));
+        return;
+      }
+    }
+
+    store!.setValue(nextValue);
+  }
+
+  function handleBlur(event: React.FocusEvent<HTMLInputElement>) {
+    restoreInputValue();
+
+    onBlur?.(event);
+  }
+
+  function handleKeyDown(event: React.KeyboardEvent<HTMLInputElement>) {
+    if (event.key === 'Enter') {
+      event.preventDefault();
+      restoreInputValue();
+    }
+    onKeyDown?.(event);
+  }
+
+  return (
+    <AriaCombobox
+      ref={ref}
+      {...props}
+      onBlur={handleBlur}
+      onKeyDown={handleKeyDown}
+      className={clsx('dbv-kit-combobox', props.className)}
+      autoSelect
+    />
+  );
+}
+
+export interface ComboboxPopoverProps extends AriaComboboxPopoverProps {
+  children?: React.ReactNode;
+}
+
+/**
+ * ComboboxPopover - Wrapper around AriaKit's popover with default styles
+ */
+export function ComboboxPopover({ children, className, ...props }: ComboboxPopoverProps) {
+  return (
+    <AriaComboboxPopover gutter={8} portal sameWidth unmountOnHide className={clsx('dbv-kit-combobox__popover', className)} {...props}>
+      {children}
+    </AriaComboboxPopover>
+  );
+}
+
+export interface ComboboxItemProps<T> extends AriaComboboxItemProps {
+  item?: T;
+  getInputFromItem?: (value: T) => string;
+}
+
+/**
+ * ComboboxItem - An option in the combobox popover
+ * Uses the search state to determine visibility
+ */
+export function ComboboxItem<T = unknown>({
+  value,
+  item,
+  ref,
+  getInputFromItem,
+  ...props
+}: ComboboxItemProps<T> & { ref?: React.Ref<HTMLDivElement> }) {
+  const store = useComboboxContext();
+  const searchValue = useStoreState(store, 'value') || '';
+  const selectedValue = useStoreState(store, 'selectedValue') || '';
+
+  const targetText = [value ?? '', item && getInputFromItem ? getInputFromItem(item) : ''].filter(Boolean).join(' ');
```

Comment:
I'm not sure that we need to filter by value if we have `item and getInputFromItem` (the user will filter by the text that he sees)